### PR TITLE
Add blob type

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

**Comment by thinca** [comment]
Blob literal should have an even number of hex characters.

```vim
echo 0zAABB     " => OK
echo 0zAABBC    " => E973: Blob literal should have an even number of hex characters
echo 0zAAB.BCC  " => E973: Blob literal should have an even number of hex characters
```

Should we make an error in this case?
How do you think?

File: test/test_err_blob.vim [hunk 1]
```diff
@@ -0,0 +1,2 @@
+function Foo(a, b, C, d_e, _f, g_, h0, i1, a:bar)
```
Comment (thinca) [inline test/test_err_blob.vim]: This is just a copy of `test_err_funcarg`, not for blob.

File: autoload/vimlparser.vim [hunk 2]
```diff
@@ -3941,6 +3951,23 @@ function! s:StringReader.read_odigit()
   return r
 endfunction
 
+function! s:StringReader.read_blob()
+  let r = ''
+  while 1
+    let s = self.peekn(2)
+    if s == ''
```
Comment (thinca) [inline autoload/vimlparser.vim]: Code may can continuous.

```vim
echo 0z00 + 0zFF
```

This case should be added to test.

**Comment by thinca** [comment]
@mattn Can we merge this?

**Comment by mattn** [comment]
Thanks your review.

**Pull request merged**